server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestShrinkIndexAction.java

```diff
@@ -47,23 +45,12 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
-        if (request.param("target") == null) {
-            throw new IllegalArgumentException("no target index");
-        }
-        if (request.param("index") == null) {
-            throw new IllegalArgumentException("no source index");
-        }
```

Comment:
Is this some sort of cleanup?

Reply:
yes that can't happen already as this is validated at the response level , no need to do it at the rest action level too.